### PR TITLE
Add stable sort operator

### PR DESCRIPTION
**Pull request by xytintel**
eg. sort.stable, sort.values_stable, argsort.stable

### CODE REVIEW

File: src/comm/SYCLHelpers.h
Comment (fengyuan14) [inline src/comm/SYCLHelpers.h:1]: What's the intention of the change in the file? Just move to lines after?

File: src/aten/sycl/Sorting.h
Comment (fengyuan14) [inline src/aten/sycl/Sorting.h:1]: Kernel interface file naming is `xxxKernels.h`.
You may put sort kernels implementation in `Sorting.h`, and put interface declaration in `SortingKernels.h`. Like,
image

File: src/aten/sycl/SortingKernels.h [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+    bool IS_DESCENDING,
+    int SUBGROUP_SIZE = 32>
+void segmented_sort_pairs_(
+    const key_t* keys_in,
+    key_t* keys_out,
+    const value_t* values_in,
+    value_t* values_out,
+    int num_segments,
+    int num_elements) {
+  if (num_elements > 4096) {
```
Comment (fengyuan14) [inline src/aten/sycl/SortingKernels.h:420]: Please give comments for these heuristics and these magic number.

File: src/aten/sycl/SortingKernels.h [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+  }
+}
+
+// ======================= interface =======================
+
+template <
+    typename key_t,
+    typename value_t,
+    bool IS_DESCENDING,
+    int SUBGROUP_SIZE = 32>
```
Comment (fengyuan14) [inline src/aten/sycl/SortingKernels.h:412]: Please add comments for hard-coded sg size. It is not platform general code. We say we agree some specific code for kernel performance on specific hardware platform. But we need some clear comments for it.

**Comment by fengyuan14** [comment]
Please remove CPU fallback.

**Comment by fengyuan14** [comment]
Please add op into fine gran case.

File: src/comm/SYCLHelpers.h
Comment (xytintel) [reply src/comm/SYCLHelpers.h:1]: Provide developers with limited choices to improve code consistency.

**Comment by xytintel** [comment]
image
image

File: src/comm/SYCLHelpers.h
Comment (fengyuan14) [reply src/comm/SYCLHelpers.h:1]: LGTM

File: test/xpu/run_test.py [hunk 3]
```diff
@@ -729,5 +732,19 @@ def launch_test(test_case, skip_list=None, exe_list=None):
 )
 res += launch_test("test_foreach_xpu.py", exe_list=execute_list)
 
+# test_sort_and_select
+skip_list = (
+    "test_isin_different_devices_xpu_float32", # AssertionError: RuntimeError not raised
```
Comment (fengyuan14) [inline test/xpu/run_test.py:737]: Please give a justification for skipping.

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**